### PR TITLE
Create README

### LINKED ISSUE #19: Create README
Need a good README that documents implementation of this reference architecture including avaialble variables and their behavior and how to import to existing configuration. We should also include the final architecture diagram of what this creates.

### PR DESCRIPTION
**Pull request by kulinacs**
Creates a README covering module usage, inputs and outputs, importing existing infrastructure, accessing the deployed infrastructure, and the current architecture diagram.

Additionally added descriptions to all variables and outputs so I/O table could be generated with terraform-docs.

Fixes #19

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -71,22 +79,26 @@ EOF
...
-  type    = "list"
-  default = ["50.58.217.226/32"]
+  description = "CIDR address allowed to access SSH on bastion server"
+  type        = "list"
+  default     = ["50.58.217.226/32"]
 }
 
 variable "app_tier_userdata" {
-  type = "string"
+  description = "Provision script for App Server"
```
Comment (sean-nixon) [inline variables.tf]: I'd adjust this to "Plaintext user data provisioning script for App Servers" to clarify it's not supposed to be base64 encoded

File: variables.tf [hunk 2]
```diff
@@ -102,6 +114,7 @@ EOF
 }
 
 variable "app_tier_ami" {
-  type    = "string"
-  default = "ami-024a64a6685d05041"
+  description = "AMI for Bastion Server (Default: HVM EBS x64 Ubuntu Bionic 18.04)"
```
Comment (sean-nixon) [inline variables.tf]: This says Bastion instead of Application Tier Server

File: variables.tf [hunk 3]
```diff
@@ -39,17 +44,20 @@ variable "private_cidrs" {
...
-  type    = "string"
-  default = "ami-0ff8a91507f77f867"
+  description = "AMI for Web Server (Default: HVM EBS x64 Amazon Linux)"
+  type        = "string"
+  default     = "ami-0ff8a91507f77f867"
 }
 
 variable "web_tier_userdata" {
-  type = "string"
+  description = "Provision script for Web Server"
```
Comment (sean-nixon) [inline variables.tf]: See comment about app tier

File: README.md [hunk 4]
```diff
@@ -1,3 +1,90 @@
...
+
+    ssh -L 5432:master-db:5432 -N bastion
+
+This will forward all traffic sent to `localhost:5432` to the database.
+
+### Web Auto Scaling Group
+
+The web auto scaling group can be accessed over ssh from the bastionhost with the specified SSH key and the default user `ec2-user`.
+
+    ssh -J ec2-user@bastion ec2-user@web
```
Comment (sean-nixon) [inline README.md]: I'm not familiar with the `-J` switch... does this require the private ssh key to first be copied to the bastion?

**Comment by sean-nixon** [comment]
This looks great! Just have a few comments (see above)

File: README.md (hunk 4 above)
Comment (kulinacs) [reply README.md]: -J switch is synonymous with the `ProxyJump` directive in ssh config. The best kind of magic, you don't need to put keys on the bastion host. My assumption is that it forwards the SSH agent, but I don't know that for sure.

**Comment by kulinacs** [comment]
Requested changes made

File: README.md (hunk 4 above)
Comment (sean-nixon) [reply README.md]: Cool beans. I’m a fan of your ssh sorcery.

**Pull request merged**